### PR TITLE
#49 Require `maximize=True` for dual optimizers

### LINKED ISSUE #49: Do `maximize=True` for dual_optimizers
## Enhancement

Pytorch optimizers include a `maximize` flag ([Pytorch Issue](https://github.com/pytorch/pytorch/issues/68052))*. When set to `True`, the sign of gradients is flipped inside `optimizer.step()` before computing parameter updates. This enables gradient ascent steps natively.

> **_NOTE:_**  This sign flip does not affect the parameter's `.grad` attribute. 

Cooper currently populates the gradients of dual variables with their *negative* value, so that descent steps performed by the dual optimizer are in fact ascent steps towards optimizing the problem formulation.

https://github.com/cooper-org/cooper/blob/09df7597e9e42deed4c7a162a2bb345868c8bf23/cooper/constrained_optimizer.py#L401-L406

We should not do the sign flipping manually but rather force set `maximize=True` when instantiating the dual optimizer.

\* This has been implemented on on Pytorch's master branch for every optimizer *but* LBFGS . On v1.12, Adam, SGD and AdaGrad support the flag, but not RMSProp. An assert could be included to ensure that the requested dual optimizer supports the flag. 

⚠ This change would break compatibility with versions of Pytorch prior to 1.12.

## Motivation

Manually flipping gradients immediately after calculating them (thus ensuring that this happens before calls to `dual_optimizer.step()`) is error prone.
Moreover, keeping track of the fact that gradients have a sign flipped is inconvenient.

By implementing this change we would adopt the official Pytorch approach for performing ascent steps.

## Alternatives

The current implementation is functional. 

## References

- https://github.com/pytorch/pytorch/issues/68052

### PR DESCRIPTION
**Pull request by juan43ramirez**
Closes #49 

## Changes

When instantiating a `dual_optimizer`, we set `maximize=True`.
Pytorch versions below 1.12 are no longer supported. `setup.cfg` and tox adjusted.

## Testing

Passing

### CODE REVIEW

**Comment by gallego-posada** [comment]
This change has already been integrated in branch [1.0.0-alpha](/github/cooper-org/cooper/tree/gallego-posada/1.0.0-alpha) (2fc418a33f955ec074c37307ffec3d4c33c4b196). Closing without merge since eventual merge from `1.0.0-alpha` into `dev` will include this changes.
